### ISSUE TITLE
Word boundary (\b) is not a good separator for symbol names

### ISSUE THREAD

**Issue by batracos** [ViktorQvarfordt/Sublime-WolframLanguage#15]
The current rules will match an expression like
```
Plus$Plus
```
as a system symbol (`Plus`) followed by a normal symbol (`$Plus`).

I have not looked into all the consequences but my current proposal is to replace the second `\b` in the regex expression by ``(?![$`[:alnum:]])``. This would also fix expressions like
```
Plus`foo
```

**Comment by ViktorQvarfordt** [comment]
How about replacing `\b` with ``[\b$`]``? In any case, isn't this best fixed by updating the variable "symbol", currently defined as

```
symbol: '[$[:alpha:]]+[$[:alnum:]]*'
```

**Comment by batracos** [comment]
In which way would you update that? I can only think of a negative look-behind on defined symbols would be terribly expensive.

``[\b$`]`` would just move the backtick or the dollar to the other match but in this case we want `Plus$Plus` to be a single scope.

@chere005, @Shigma, any idea?

**Comment by batracos** [comment]
I did not touch this because good patterns (that I was able to find) bring bad performance and vice versa.

**Comment by Shigma** [comment]
> I did not touch this because good patterns (that I was able to find) bring bad performance and vice versa.

I don't consider it a problem since the current syntax for wolfram is not complicated compared with other languages.

In my option, how about inserting a pattern ``(?![$`])`` before the ending `\b`?

**Comment by batracos** [comment]
This is exactly what I am using locally. I never committed that because I am not super sure about it but given that we are already in two people coming up with the same solution I would go for it.
```
\b(?:AASTriangle|...)(?![$`])\b
```

Regarding the contexts: I don't think that `Image` should be highlighted in `` Image`foo ``.

**Comment by chere005** [comment]
I just tested ```(?![$`])``` and it looks good to me, @Shigma can you please add this to your current PR?

**Comment by Shigma** [comment]
> I just tested `` (?![$`]) `` and it looks good to me, @Shigma can you please add this to your current PR?

It has already been included in one of @batracos ' branches, see [here](https://github.com/batracos/Sublime-WolframLanguage/commit/5822b8c8c19f4549651d49d94e5db7d2dfaae78b#diff-f45849230713269b1b695c0594ea22d7R24).

**Comment by batracos** [comment]
Yeah sorry. I have been meaning to create a PR for that for some time. I'll do it today.
